handler/openid/flow_device_auth.go

```diff
@@ -0,0 +1,55 @@
+// Copyright © 2024 Ory Corp
+// SPDX-License-Identifier: Apache-2.0
+
+package openid
+
+import (
+	"context"
+
+	"github.com/ory/fosite/handler/rfc8628"
+
+	"github.com/ory/fosite"
+	"github.com/ory/x/errorsx"
+)
+
+// OpenIDConnectDeviceHandler a response handler for the Device Authorization Grant with OpenID Connect identity layer
+type OpenIDConnectDeviceHandler struct {
+	OpenIDConnectRequestStorage   OpenIDConnectRequestStorage
+	OpenIDConnectRequestValidator *OpenIDConnectRequestValidator
+	DeviceCodeStrategy            rfc8628.DeviceCodeStrategy
+
+	Config interface {
+		fosite.IDTokenLifespanProvider
+	}
+
+	*IDTokenHandleHelper
+}
+
+func (c *OpenIDConnectDeviceHandler) HandleDeviceEndpointRequest(ctx context.Context, dar fosite.DeviceRequester, resp fosite.DeviceResponder) error {
+	if !(dar.GetGrantedScopes().Has("openid")) {
+		return nil
+	}
+
+	if !dar.GetClient().GetGrantTypes().Has(string(fosite.GrantTypeDeviceCode)) {
+		return nil
+	}
+
+	if len(resp.GetDeviceCode()) == 0 {
+		return errorsx.WithStack(fosite.ErrMisconfiguration.WithDebug("The device code has not been issued yet, indicating a broken code configuration."))
+	}
+
+	if err := c.OpenIDConnectRequestValidator.ValidatePrompt(ctx, dar); err != nil {
+		return err
+	}
```

Comment:
Not sure if we want this, a `prompt` param is not defined in the device flow spec (it makes sense as it is an oauth2 rfc and `prompt` is an oidc param).

I took a look at how this is handled by google, microsoft and amazon and they do not talk about a `prompt` param being supported in their docs. I would suggest to remove this